packages/host/app/lib/browser-queue.ts

```diff
@@ -59,7 +59,7 @@ export class BrowserQueue implements Queue {
     if (this.isDestroyed) {
       throw new Error(`Cannot publish job on a destroyed Queue`);
     }
```

Comment:
is there supposed to be a guard rail for publishing to an unregistered category??

Reply:
You are allowed to publish to an unregistered category. in fact we have a test that does that. the idea is that the job will sit idle until a handler registers a category that the jobs uses. This prevents ordering issues when starting up as handlers register for various categories while at the same time jobs may be published to the queue.

Reply:
its actually the very first test in `queue-test.ts`:
```ts
'it can run a job': async (assert, { queue }) => {
    let job = await queue.publish<number>('increment', 17, {
      queueName: 'increment-queue',
    });
    queue.register('increment', async (a: number) => a + 1);
    let result = await job.done;
    assert.strictEqual(result, 18);
  },
```

Note how the job is published first and the handler is registered second.

Reply:
> its actually the very first test in `queue-test.ts`:
> 
> ```ts
> 'it can run a job': async (assert, { queue }) => {
>     let job = await queue.publish<number>('increment', 17, {
>       queueName: 'increment-queue',
>     });
>     queue.register('increment', async (a: number) => a + 1);
>     let result = await job.done;
>     assert.strictEqual(result, 18);
>   },
> ```
> 
> Note how the job is published first and the handler is registered second.

Noise real useful to know thx